lib/puppet/util/windows/user.rb

```diff
@@ -57,9 +73,9 @@ def check_token_membership
   end
   module_function :check_token_membership
 
-  def password_is?(name, password)
+  def password_is?(name, domain, password)
```

Comment:
A more backwards-compatible approach can be to have the domain as the last parameter of the `password_is` method and to make it optional (default to '.'). Just a thought. Same for logon_user if you decide to do it.

Reply:
That was my first instinct also, but got in some trouble (can't remember exactly what) with `logon_user` method because of the `&block` parameter and I didn't want to have inconsistency in the the order of the parameters for these methods (some having the domain as the last parameter and some not). Will try looking further into it.

Reply:
For `logon_user` it should work if you have `(name, password, domain = '.', &block)` in the method definition.

Reply:
I agree with @GabrielNagy  here, In order not no break others and i would go for having `domain` as a third parameter with the `'.'` default